.github/workflows/publish-to-pypi.yml

```diff
@@ -24,6 +24,37 @@ jobs:
           --sdist
           --wheel
           --outdir dist/
+      - name: Publish distribution 📦 to PyPI
+        if: startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN_OPTIMAGIC }}
+  build-n-publish-estimagic:
+    name: Build and publish estimagic Python 🐍 distributions 📦 to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install pypa/build
+        run: >-
+          python -m
+          pip install
+          build
+          toml
+          --user
+      - name: Set name entry in pyproject.toml to estimagic
+        run: >-
+          python .scripts/update-name-in-pyproject.py
```

Comment:
After starting to work on the enhancement proposals I see the advantage of having `optimagic` and `estimagic` in one package. I think we should completely abandon plans to split at some time in the future. NumFocus is against it anyways. Then we also don't have to make double releases for very long.

I would therefore say, that we should add a `FutureWarning` to `estimagic.__init__.py` and `optimagic.__init__.py` when we release the `estimagic` package. The future warning could say something like: 

```txt
"estimagic has been renamed to optimagic. Please uninstall estimagic and install optimagic 
instead. Don't worry, your estimagic imports will still work if you install optimagic and simple
warnings will help you to adjust them for future releases.  If you use pip, the commands are
[please add uninstall and reinstall commands]. If you use conda or mamba, the commands
are [...]."
```